src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -102,16 +109,54 @@ public async Task SpecializeAsync()
         {
             _logger.LogInformation("Starting language worker channel specialization");
             _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+
             ILanguageWorkerChannel languageWorkerChannel = await GetChannelAsync(_workerRuntime);
             if (_workerRuntime != null && languageWorkerChannel != null)
             {
-                _logger.LogInformation("Loading environment variables for runtime: {runtime}", _workerRuntime);
-                await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
+                // Don't use placeholder process if app is not read-only or important environment config has changed
+                // TODO: Allow non read-only apps to use placeholder process - https://github.com/Azure/azure-functions-host/issues/4534
+                if (UsePlaceholderProcess(_workerRuntime))
+                {
+                    _logger.LogInformation("Loading environment variables for runtime: {runtime}", _workerRuntime);
```

Comment:
change to log debug. Add logging here, valid placeholdermode process and also in the else block, incompatible palceholdermode process,